projects/core/directives/icons/icons.directive.ts

```diff
@@ -34,22 +39,31 @@ class TuiIconsStyles {}
     standalone: true,
     host: {
         tuiIcons: '',
-        '[style.--t-icon-start]': 'iconStartUrl()',
-        '[style.--t-icon-end]': 'iconEndUrl()',
+        '[style.--t-icon-start]': 'iconStartResource()',
+        '[style.--t-icon-end]': 'iconEndResource()',
+        '[attr.data-icon-start]': 'iconStartMode()',
+        '[attr.data-icon-end]': 'iconEndMode()',
     },
 })
 export class TuiIcons {
     private readonly resolver: TuiStringHandler<string> = tuiInjectIconResolver();
 
     protected readonly nothing = tuiWithStyles(TuiIconsStyles);
-    protected readonly iconStartUrl = computed(() => this.resolve(this.iconStart()));
-    protected readonly iconEndUrl = computed(() => this.resolve(this.iconEnd()));
+    protected readonly iconStartResource = computed(() => this.resolve(this.iconStart()));
+    protected readonly iconEndResource = computed(() => this.resolve(this.iconEnd()));
+    protected readonly iconStartMode = computed(() =>
+        tuiGetIconMode(this.iconStart()?.toString()),
+    );
+
+    protected readonly iconEndMode = computed(() =>
+        tuiGetIconMode(this.iconEnd()?.toString()),
+    );
 
     public readonly iconStart = signal<Icon>(
         inject(TUI_ICON_START, {self: true, optional: true}) || '',
     );
 
-    public readonly iconEnd = signal<string>(
+    public readonly iconEnd = signal<Icon>(
```

Comment:
It is supposed to be a string, the `Icon` mode is a hack to make this directive compatible with `Avatar`